### PR TITLE
Try to unbreak CI by xfailing OSX Tk tests

### PR DESCRIPTION
**Pull request by timhoffm**
Stopgap solution for #23094.

Mid-term we should get the CI environment sorted out to have a working Tk setup. But until then, it's better to xfail than to have all PRs cluttered with these failures.

This is a blind attempt. - CI will decide whether I did everything correctly.

### CODE REVIEW

File: lib/matplotlib/tests/test_backend_tk.py [hunk 1]
```diff
@@ -37,6 +37,10 @@ def _isolated_tk_test(success_count, func=None):
         sys.platform == "linux" and not _c_internal_utils.display_is_valid(),
         reason="$DISPLAY and $WAYLAND_DISPLAY are unset"
     )
+    @pytest.xfail(  # GitHub issue #23094
+        reason="Tk version mismatch on OSX CI",
+        condition=sys.platform == 'darwin'
+    )
```
Comment (oscargus) [inline lib/matplotlib/tests/test_backend_tk.py:43]:
```suggestion
    @pytest.xfail(  # GitHub issue #23094
        sys.platform == 'darwin',
        reason="Tk version mismatch on OSX CI"
    )
```

Comment (oscargus) [reply lib/matplotlib/tests/test_backend_tk.py:43]: Commiting this to confirm that it fixes it. Feel free to force push without lease to get rid of it if preferred.

File: lib/matplotlib/tests/test_backend_tk.py [hunk 2]
```diff
@@ -37,6 +37,10 @@ def _isolated_tk_test(success_count, func=None):
         sys.platform == "linux" and not _c_internal_utils.display_is_valid(),
         reason="$DISPLAY and $WAYLAND_DISPLAY are unset"
     )
+    @pytest.xfail(  # GitHub issue #23094
```
Comment (oscargus) [inline lib/matplotlib/tests/test_backend_tk.py:40]:
```suggestion
    @pytest.mark.xfail(  # GitHub issue #23094
```

**Comment by timhoffm** [comment]
@oscargus thanks for fixing this! Yes, your fix is is correct and the first commit is also good.

@all I suggest to squash-merge this.

**Review by anntzer — APPROVED** [review]

**Pull request merged**